source/data/js_api.yml

```diff
@@ -16,6 +15,23 @@
   - attr: "configParams"
     desc: "configParams is OPTIONAL, It is the set of parameters to enable/disable certain features in the library. See [Integrating with your SDK - Step 2](#step-2-initialize) section below for details. "
 
+- api: initializeWithTokenGenrator()
+  desc: "authenticates with the callstats.io backend to setup a trusted relationship with it using third party authentication. See [integration section](#integrating-with-your-sdk) for details on when to call this API."
+  status: "implemented"
+  params:
+  - attr: "AppID"
+    desc: "of type _String_. It is obtained from callstats.io backend; discussed in a [later section](#preparing-the-origin-server)."
+  - attr: "TokenGeneratorCallback"
+    desc: "of type _function(forcenew, callback)_. It is obtained from callstats.io backend; discussed in a [later section](#preparing-the-origin-server)."
```

Comment:
secondly, if the AppSecret is string then old auth is used, if it is a function, the new auth is used?